### PR TITLE
DEMO-388 (feature): input source state implementation and form control

### PR DESCRIPTION
**Pull request by NicDominguez**
**JIRA Ticket [DEMO-388](https://rocketcom.atlassian.net/browse/DEMO-388)**

## Brief Description

This PR incorporates the use of the currentTrackFile state into the input-source form under the Propagator Utility. 
The form will modify the track file with a new ephemeris file using the fields available. I also built the generation of ephemerides data so they can be displayed in the view orbit table view. This ephemerides data should replace the sat cords but for now it has no relation until we know the proper static coordinates we want to use for the displayed orbit. 

## Issues and/or Limitations

Without specific direction from design I made up much of the functionality of the form and its interaction with the table. There is a bug with the table not updating with changed state that I thought out of scope for this PR. 

I left out updating the Orbit Source fields to avoid overlap with another PR which touches on the functionality.

## Final Checklist

- [X] Works in Chrome
- [X] Works in Firefox
- [X] Works in Safari
- [X] Handles responsiveness as required
- [X] Dark theme styles are correct
- [x] Light theme styles are correct

### CODE REVIEW

File: src/app/mock-data/generate-data.ts [hunk 1]
```diff
@@ -77,32 +79,70 @@ const generateTrackFile = (
...
   trackFileId: string,
-  index: number,
+  startDatetime: Date,
+  span: number,
   satPos1X: number,
   satPos1Y: number,
   satPos2X: number,
   satPos2Y: number
 ): EphemerisFile => {
+  console.log(startDatetime);
```
Comment (Paterma) [inline src/app/mock-data/generate-data.ts:91]: Remove console.logs please

**Review by Paterma — CHANGES_REQUESTED** [review]
Looks really good! 
Maybe we can update the toast for Primary Action to say like "See changes settings in View Orbit Tab" or something along those lines. It wouldn't be super clear to me that that is where you'd see the changes unless I knew where I was going. 
Great work though, just remove those console.logs :)

File: src/app/mock-data/generate-data.ts (hunk 1 above)
Comment (NicDominguez) [reply src/app/mock-data/generate-data.ts:91]: Good catch

File: src/app/main/utility-components/propagator-utility/input-source/input-source.component.ts [hunk 2]
```diff
@@ -2,17 +2,119 @@ import { Component } from '@angular/core';
...
+        message: 'Please provide required values',
+        hideClose: false,
+        closeAfter: 3000,
+      });
+      return;
+    }
+    if (this.currentTrackFile === null) return;
+    if (!this.sourceSettingsForm.value.epoch) return;
+
+    console.log(this.sourceSettingsForm);
```
Comment (Paterma) [inline src/app/main/utility-components/propagator-utility/input-source/input-source.component.ts:80]: One more here

**Review by Paterma — APPROVED** [review]

**Pull request merged**